File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        ans = []
        p = q = 0
        for a, b in zip(A, B):
            p |= 1 << a
            q |= 1 << b
            ans.append((p & q).bit_count())
        return ans

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.js
/**
 * @param {number[]} A
 * @param {number[]} B
 * @return {number[]}
 */
var findThePrefixCommonArray = function(A, B) {
  const n = A.length;
  const count = new Array(n + 1).fill(0);
  const ans = new Array(n).fill(0);
  let cnt = 0;
  
  for (let i = 0; i < n; i++) {
    if (++count[A[i]] === 2) cnt++;
    if (++count[B[i]] === 2) cnt++;
    
    ans[i] = cnt;
  }
  
  return ans;
};